### ISSUE TITLE
cannot create class 'liveplugin.toolwindow.addplugin.AddNewKotlinPluginAction' [Plugin: LivePlugin]

### ISSUE THREAD

**Issue by Stranger2015** [dkandalov/live-plugin#81]
cannot create class 'liveplugin.toolwindow.addplugin.AddNewKotlinPluginAction' [Plugin: LivePlugin]
![2018-04-29_21-50-36](https://user-images.githubusercontent.com/16267580/39409944-0b330dae-4bf8-11e8-9515-f7b2663d40f7.png)

**Comment by Stranger2015** [comment]
I just decided to attach screenshot. no reproducton steps, no clues anymore

**Comment by YonathanB** [comment]
I have the same  error in webstorm 2017.3

**Comment by Stranger2015** [comment]
First I thought the cause of error is the toolbar being off but later it  was not confirmed

**Comment by krasa** [comment]
Maybe it's the same thing as with IJ 2018.2 from current master. 
They deleted bunch of icons, totally breaking my plugins... Except UI designer does not even log it right:
```
java.lang.NullPointerException
	at javax.swing.ImageIcon.<init>(ImageIcon.java:217)
	at krasa.formatter.plugin.ProjectSettingsForm.$$$setupUI$$$(ProjectSettingsForm.java)
	at krasa.formatter.plugin.ProjectSettingsForm.<init>(ProjectSettingsForm.java:144)
```


This is yours:
```
cannot create class 'liveplugin.toolwindow.popup.NewKotlinFileAction' [Plugin: LivePlugin]
com.intellij.diagnostic.PluginException: cannot create class 'liveplugin.toolwindow.popup.NewKotlinFileAction' [Plugin: LivePlugin]
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.error(ActionManagerImpl.java:184)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.convertStub(ActionManagerImpl.java:155)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.getActionImpl(ActionManagerImpl.java:479)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.getAction(ActionManagerImpl.java:467)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.preloadActions(ActionManagerImpl.java:1278)
	at com.intellij.openapi.actionSystem.impl.ActionPreloader.preload(ActionPreloader.java:32)
	at com.intellij.openapi.application.Preloader.lambda$null$0(Preloader.java:74)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$1(CoreProgressManager.java:157)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:580)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:525)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:85)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:144)
	at com.intellij.openapi.application.Preloader.lambda$initComponent$1(Preloader.java:72)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:218)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at com.intellij.util.ExceptionUtil.rethrow(ExceptionUtil.java:160)
	at com.intellij.util.ReflectionUtil.createAsDataClass(ReflectionUtil.java:489)
	at com.intellij.util.ReflectionUtil.newInstance(ReflectionUtil.java:422)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.convertStub(ActionManagerImpl.java:143)
	... 15 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at com.intellij.util.ReflectionUtil.createAsDataClass(ReflectionUtil.java:465)
	... 17 more
Caused by: java.lang.RuntimeException: Can't find icon in '/org/jetbrains/kotlin/idea/icons/kotlin_file.png' near class liveplugin.IdeUtil$KotlinScriptFileType$Companion$kotlinScriptIcon$2
	at com.intellij.openapi.util.IconLoader.findIcon(IconLoader.java:191)
	at com.intellij.openapi.util.IconLoader.findIcon(IconLoader.java:176)
	at com.intellij.openapi.util.IconLoader.findIcon(IconLoader.java:171)
	at com.intellij.openapi.util.IconLoader.findIcon(IconLoader.java:138)
	at liveplugin.IdeUtil$KotlinScriptFileType$Companion$kotlinScriptIcon$2.invoke(IdeUtil.kt:247)
	at liveplugin.IdeUtil$KotlinScriptFileType$Companion$kotlinScriptIcon$2.invoke(IdeUtil.kt:243)
	at kotlin.SynchronizedLazyImpl.getValue(Lazy.kt:131)
	at liveplugin.IdeUtil$KotlinScriptFileType$Companion.getKotlinScriptIcon(IdeUtil.kt)
	at liveplugin.IdeUtil$KotlinScriptFileType$Companion.access$getKotlinScriptIcon$p(IdeUtil.kt:243)
	at liveplugin.IdeUtil$KotlinScriptFileType.getIcon(IdeUtil.kt:238)
	at liveplugin.toolwindow.popup.NewFileAction.<init>(NewFileAction.java:36)
	at liveplugin.toolwindow.popup.NewKotlinFileAction.<init>(new-element-actions.kt:21)
	... 22 more

```

**Comment by Stranger2015** [comment]
I didn't touch neither IDEA's nor liveplugin's sources (if you meant that.

**Comment by krasa** [comment]
Well, my error is probably unrelated, I will create a new issue.
BTW next time click on details and post the stacktrace :]

**Comment by krasa** [comment]
Well, I cannot reproduce my error now either...

**Comment by Stranger2015** [comment]
Looks like today is the "next-time" =)

Stack trace 1 of 2
cannot create class 'liveplugin.toolwindow.addplugin.AddNewKotlinPluginAction' [Plugin: LivePlugin]
com.intellij.diagnostic.PluginException: cannot create class 'liveplugin.toolwindow.addplugin.AddNewKotlinPluginAction' [Plugin: LivePlugin]
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.error(ActionManagerImpl.java:184)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.convertStub(ActionManagerImpl.java:155)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.getActionImpl(ActionManagerImpl.java:479)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.getAction(ActionManagerImpl.java:467)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.preloadActions(ActionManagerImpl.java:1278)
	at com.intellij.openapi.actionSystem.impl.ActionPreloader.preload(ActionPreloader.java:32)
	at com.intellij.openapi.application.Preloader.lambda$null$0(Preloader.java:74)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$1(CoreProgressManager.java:157)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:580)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:525)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:85)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:144)
	at com.intellij.openapi.application.Preloader.lambda$initComponent$1(Preloader.java:72)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:212)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at com.intellij.util.ExceptionUtil.rethrow(ExceptionUtil.java:174)
	at com.intellij.util.ReflectionUtil.createAsDataClass(ReflectionUtil.java:489)
	at com.intellij.util.ReflectionUtil.newInstance(ReflectionUtil.java:422)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.convertStub(ActionManagerImpl.java:143)
	... 15 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at com.intellij.util.ReflectionUtil.createAsDataClass(ReflectionUtil.java:465)
	... 17 more
Caused by: java.lang.NoClassDefFoundError: org/jetbrains/kotlin/codegen/CompilationException
	at liveplugin.toolwindow.addplugin.AddNewKotlinPluginAction.<init>(AddNewPluginAction.kt:29)
	... 22 more
Caused by: java.lang.ClassNotFoundException: org.jetbrains.kotlin.codegen.CompilationException PluginClassLoader[LivePlugin, 0.6.1 beta] com.intellij.ide.plugins.cl.PluginClassLoader@6de23f4
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:63)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 23 more

Stack trace 2 of 2 and,  further,  repeating `CompilationException`

org/jetbrains/kotlin/codegen/CompilationException
java.lang.NoClassDefFoundError: org/jetbrains/kotlin/codegen/CompilationException
	at liveplugin.toolwindow.addplugin.AddNewKotlinPluginAction.<init>(AddNewPluginAction.kt:29)
	at liveplugin.toolwindow.PluginToolWindowManager$PluginToolWindow$Companion.createAddPluginsGroup(PluginToolWindowManager.kt:259)
	at liveplugin.toolwindow.PluginToolWindowManager$PluginToolWindow$Companion.access$createAddPluginsGroup(PluginToolWindowManager.kt:158)
	at liveplugin.toolwindow.PluginToolWindowManager$PluginToolWindow.createToolBar(PluginToolWindowManager.kt:132)
	at liveplugin.toolwindow.PluginToolWindowManager$PluginToolWindow.createContent(PluginToolWindowManager.kt:113)
	at liveplugin.toolwindow.PluginToolWindowManager$PluginToolWindow.registerWindowFor(PluginToolWindowManager.kt:97)
	at liveplugin.toolwindow.PluginToolWindowManager$init$1.projectOpened(PluginToolWindowManager.kt:69)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.util.messages.impl.MessageBusConnectionImpl.deliverMessage(MessageBusConnectionImpl.java:117)
	at com.intellij.util.messages.impl.MessageBusImpl.doPumpMessages(MessageBusImpl.java:426)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpWaitingBuses(MessageBusImpl.java:387)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:376)
	at com.intellij.util.messages.impl.MessageBusImpl.sendMessage(MessageBusImpl.java:357)
	at com.intellij.util.messages.impl.MessageBusImpl.access$200(MessageBusImpl.java:43)
	at com.intellij.util.messages.impl.MessageBusImpl$2.invoke(MessageBusImpl.java:208)
	at com.sun.proxy.$Proxy21.projectOpened(Unknown Source)
	at com.intellij.openapi.project.impl.ProjectManagerImpl.fireProjectOpened(ProjectManagerImpl.java:761)
	at com.intellij.openapi.project.impl.ProjectManagerImpl.lambda$null$6(ProjectManagerImpl.java:386)
	at com.intellij.openapi.application.TransactionGuardImpl.lambda$submitTransactionAndWait$2(TransactionGuardImpl.java:165)
	at com.intellij.openapi.application.TransactionGuardImpl.runSyncTransaction(TransactionGuardImpl.java:88)
	at com.intellij.openapi.application.TransactionGuardImpl.lambda$submitTransaction$1(TransactionGuardImpl.java:111)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.doRun(LaterInvocator.java:447)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:431)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:415)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:779)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:720)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:395)
	at com.intellij.ide.IdeEventQueue.pumpEventsForHierarchy(IdeEventQueue.java:862)
	at com.intellij.openapi.progress.util.ProgressWindow.startBlocking(ProgressWindow.java:215)
	at com.intellij.openapi.progress.util.ProgressWindow.startBlocking(ProgressWindow.java:201)
	at com.intellij.openapi.application.impl.ApplicationImpl.runProcessWithProgressSynchronously(ApplicationImpl.java:579)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcessWithProgressSynchronously(CoreProgressManager.java:446)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.runProcessWithProgressSynchronously(ProgressManagerImpl.java:109)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runSynchronously(CoreProgressManager.java:318)
	at com.intellij.openapi.progress.impl.CoreProgressManager.run(CoreProgressManager.java:303)
	at com.intellij.openapi.progress.ProgressManager.run(ProgressManager.java:192)
	at com.intellij.openapi.project.impl.ProjectManagerImpl.loadAndOpenProject(ProjectManagerImpl.java:478)
	at com.intellij.ide.impl.ProjectUtil.openProject(ProjectUtil.java:191)
	at com.intellij.ide.RecentProjectsManagerImpl.doOpenProject(RecentProjectsManagerImpl.java:63)
	at com.intellij.ide.RecentProjectsManagerBase.doReopenLastProject(RecentProjectsManagerBase.java:677)
	at com.intellij.ide.RecentProjectsManagerBase$MyAppLifecycleListener.appStarting(RecentProjectsManagerBase.java:719)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.util.messages.impl.MessageBusConnectionImpl.deliverMessage(MessageBusConnectionImpl.java:117)
	at com.intellij.util.messages.impl.MessageBusImpl.doPumpMessages(MessageBusImpl.java:426)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpWaitingBuses(MessageBusImpl.java:387)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:376)
	at com.intellij.util.messages.impl.MessageBusImpl.sendMessage(MessageBusImpl.java:357)
	at com.intellij.util.messages.impl.MessageBusImpl.access$200(MessageBusImpl.java:43)
	at com.intellij.util.messages.impl.MessageBusImpl$2.invoke(MessageBusImpl.java:208)
	at com.sun.proxy.$Proxy46.appStarting(Unknown Source)
	at com.intellij.idea.IdeaApplication$IdeStarter.lambda$main$1(IdeaApplication.java:361)
	at com.intellij.openapi.application.TransactionGuardImpl.runSyncTransaction(TransactionGuardImpl.java:88)
	at com.intellij.openapi.application.TransactionGuardImpl.lambda$submitTransaction$1(TransactionGuardImpl.java:111)
	at com.intellij.openapi.application.TransactionGuardImpl.submitTransaction(TransactionGuardImpl.java:120)
	at com.intellij.openapi.application.TransactionGuard.submitTransaction(TransactionGuard.java:122)
	at com.intellij.idea.IdeaApplication$IdeStarter.main(IdeaApplication.java:359)
	at com.intellij.idea.IdeaApplication.lambda$run$0(IdeaApplication.java:199)
	at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:195)
	at com.intellij.idea.IdeaApplication.run(IdeaApplication.java:199)
	at com.intellij.idea.MainImpl$1.lambda$null$0(MainImpl.java:49)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:374)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: java.lang.ClassNotFoundException: org.jetbrains.kotlin.codegen.CompilationException PluginClassLoader[LivePlugin, 0.6.1 beta] com.intellij.ide.plugins.cl.PluginClassLoader@6de23f4
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:63)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)

**Comment by Stranger2015** [comment]
Kotlin plugin was disabled.  I also attached 3 screenshots with the plugins being installed at my PC.
![2018-05-13_21-39-36](https://user-images.githubusercontent.com/16267580/39970656-eb420db4-56f7-11e8-87ab-9807990e823e.png)
![2018-05-13_21-41-31](https://user-images.githubusercontent.com/16267580/39970657-eb703b8a-56f7-11e8-9577-49b9c7a70a16.png)
![2018-05-13_21-43-15](https://user-images.githubusercontent.com/16267580/39970658-eba8fe70-56f7-11e8-98ed-fcb6266cf407.png)

**Comment by Stranger2015** [comment]
Good news! After updating Kotlin plugin LivePlugin errors are non-reproducible anymore (Kotlin plugin either enabled or disabled). 
So, I'm going to close the issue today.
if somebody has to tell something about that please drop me a line in a comment.

**Comment by Stranger2015** [comment]
Good news! After updating Kotlin plugin LivePlugin errors are non-reproducible anymore (Kotlin plugin either enabled or disabled). 
So, I'm going to close the issue today.
if somebody has to tell something about that please drop me a line in a comment.

**Comment by dkandalov** [comment]
As a note, the root cause for this issue was fixed in v0.6.3.

If kotlin plugin is disabled, liveplugins written in kotlin should still work but won't get any syntax highlighting.
